packages/core/contracts/financial-templates/funding-rate-store/implementation/FundingRateStore.sol

```diff
@@ -101,33 +120,97 @@ contract FundingRateStore is FundingRateStoreInterface, Testable, Lockable {
         }
     }
 
+    /**
+     * @notice Computes the funding rate fees that a contract should pay for a period.
+     * @dev The late penalty is similar to the funding rate fee in that is is charged per second over the period
+     * between startTime and endTime.
+     *
+     * The late penalty percentage increases over time as follows:
+     *
+     * - 0-1 week since startTime: no late penalty
+     *
+     * - 1-2 weeks since startTime: 1x late penalty percentage is applied
+     *
+     * - 2-3 weeks since startTime: 2x late penalty percentage is applied
+     *
+     * - ...
+     *
+     * @param startTime defines the beginning time from which the fee is paid.
+     * @param endTime end time until which the fee is paid.
+     * @param pfc "profit from corruption", or the maximum amount of margin currency that a
+     * token sponsor could extract from the contract through corrupting the price feed in their favor.
+     * @return fundingRateFee amount owed for the duration from start to end time for the given pfc.
+     * @return latePenalty penalty percentage, if any, for paying the fee after the deadline.
+     */
+    function computeFundingRateFee(
+        uint256 startTime,
+        uint256 endTime,
+        FixedPoint.Unsigned calldata pfc
+    )
+        external
+        override
+        view
+        returns (FixedPoint.Unsigned memory fundingRateFee, FixedPoint.Unsigned memory latePenalty)
+    {
+        uint256 timeDiff = endTime.sub(startTime);
+
+        // Multiply by the unscaled `timeDiff` first, to get more accurate results.
+        fundingRateFee = pfc.mul(timeDiff).mul(fixedFundingRateFeePerSecondPerPfc);
```

Comment:
We should consider refactoring `computeFundingRateFee` and `computeRegularFee` into a library as these are mostly stateless methods